packages/dds/matrix/src/test/matrix.undo.spec.ts

```diff
@@ -574,29 +574,41 @@ describe("Matrix", () => {
 					[2, 3],
 				]);
 
+				// now undo both inserts
 				undo2.undoOperation();
 				await expect([[0, 1]]);
 
 				undo1.undoOperation();
 				await expect([]);
 
+				// redo re-inserts in opposite order to their original
+				// client 2, then client 1
 				undo2.redoOperation();
 				await expect([[2, 3]]);
 
 				undo1.redoOperation();
 				await expect([
-					[0, 1],
 					[2, 3],
```

Comment:
while this is a change is behavior, it was possible before if the order of undo/redos was reversed. each undo redo only takes into account local changes, so last one wins. here we sync in `expect` after each undo/redo, so clients local undo/redos take into account the remote changes. since the matrix was empty, these rows end up on the end of tree segment in merge-tree, so always try to insert at the end, this is expected with local reference undo/redo, rather than deleted segment undo/redo. I also modified the next part, which i think matches the intent of this test more and illustrates how the other order can happen depending on sequencing.